addons/website_version/models/experiment.py

```diff
@@ -0,0 +1,300 @@
+                        return False
+        return True
+
+    @api.multi
+    def _check_website(self):
+        for exp in self:
+            for exp_ver in exp.experiment_version_ids:
+                if not exp_ver.version_id.website_id.id == exp.website_id.id:
+                    return False
+        return True
+
+    _constraints = [
+        (_check_view, _('This experiment contains a view which is already used in another running experience'), ['state']),
+        (_check_website, _('This experiment must have versions which are in the same website'), ['website_id', 'experiment_version_ids']),
+    ]
+
+    _group_by_full = {
+        'state': lambda *args, **kwargs: ([s[0] for s in [('running', 'Running'), ('paused', 'Paused'), ('ended', 'Ended')]], dict()),
+    }
+
+    def _get_version_number(self):
+        for exp in self:
+            exp.version_number = len(exp.experiment_version_ids) + 1
+
+    name = fields.Char(string="Title", required=True)
+    experiment_version_ids = fields.One2many('website_version.experiment_version', 'experiment_id', string="experiment_version_ids")
+    website_id = fields.Many2one('website', string="Website", required=True)
+    state = fields.Selection([('running', 'Running'), ('paused', 'Paused'), ('ended', 'Ended')], 'Status', required=True, copy=False, track_visibility='onchange', default='running')
+    goal_id = fields.Many2one('website_version.goals', string="Objective", required=True)
+    color = fields.Integer('Color Index')
+    version_number = fields.Integer(compute=_get_version_number)
+    sequence = fields.Integer('Sequence', required=True, help="Test.", default=1)
+    google_id = fields.Char(string="Google_id")
+
+    @api.model
+    def create(self, vals):
+        print vals
+        exp = {
+            'name': vals['name'],
+            'objectiveMetric': self.env['website_version.goals'].browse([vals['goal_id']])[0].google_ref,
+            'status': vals['state'],
+            'variations': [{'name': 'master', 'url': 'http://localhost/master'}]
+        }
+        version_list = vals.get('experiment_version_ids', [])
+        for version in version_list:
+            name = self.env['website_version.version'].browse([version[2]['version_id']])[0].name
+            #We must give a URL for each version in the experiment
+            exp['variations'].append({'name': name, 'url': 'http://localhost/' + name})
+        if not version_list:
+            raise Warning(_("You must select at least one version in your experiment."))
+        google_id = self.env['google.management'].create_an_experiment(exp, vals['website_id'])
+        if not google_id:
+            raise Warning(_("Please verify you give the authorizations to use google analytics api ..."))
```

Comment:
I don't know if you answered on the commit; should'nt create_an_experiment crash about authorization / google analytics api ? Because here you basically think that not google_id = authorization issue. I think it will be misleading to users having an error not necessarily linked to credentials.